x/concentrated-liquidity/README.md

```diff
@@ -872,6 +872,115 @@ Then, we either proceed to the next swap step or finalize the swap.
 Once the swap is completed, we persiste the swap state to the global state
 (if mutative action is performed) and return the `amountCalculated` to the user.
 
+## Liquidity depths calculation
+
+### Calculating liquidity for buckets
+Each bucket (the area between two initialized ticks) contains ceratin amount of liquidity. The liquidity amount can be obtained through `GetTickLiquidityNetInDirection` query. `GetTickLiquidityNetInDirection` returns two results:
+- `expectedStartTickLiquidity` which is the global liquidity, the cumulative liquidity from the bucket of the current price
+- `expectedLiquidityDepths` which is list of liquidity deltas for each and every initialized tick for the full price range in a certain direction that should be cumulatively added to or subtracted starting from the global liquidity in order to get the liquidity for an arbitrary range.
+
+Querying in two different directions is possible, one being 'one for zero', which is in the direction of max tick, the other direction being 'zero for one', which is in the direction of min tick.
+
+If the direction is zero for one, the liquidity for a bucket in tick can be calculated as follows: L<sub>t</sub> = L<sub>t-1</sub> - ΔL<sub>t</sub>, t=1,2, ...
+
+If the direction is one for zero, the liquidity for the bucket in the designated tick can be calculated as follows: L<sub>t</sub> = L<sub>t-1</sub> + ΔL<sub>t</sub>, t=1,2
+
+where L<sub>0</sub> is the global(cumulative liquidity).
+
+### Deducing the quantity of tokens X and Y for a tick range
+Having obtained the liquidity depths for each liquidity buckets in the pool, we can derive an equation to calculate the quantity of each token locked for a certain price range. Let *i* and *j* be the indexes of lower and upper tick boundaries of the range we want to calculate, let P<sub>0</sub> be current price and P<sub>a</sub>, P<sub>b</sub> prices for lower tick and upper tick respectively, where P<sub>a</sub>, P<sub>b</sub> are defined as the following:
+<p align="center"> 
+P<sub>a</sub> = 1.0001<sup>i</sup>, P<sub>b</sub> = 1.0001<sup>j</sup>
+</p>
+
+Let *L* be the total liquidity locked within price range [P<sub>a</sub>, P<sub>b</sub>]. The real reserve curve with liquidity *L*  is:
+
+$(x + \frac{L}{\sqrt{P_b}})(y + L\sqrt{P_a}) = L^2$
+
+Given the amount of liquidity, the quantity of tokens available in the price range [P<sub>a</sub>, P<sub>b</sub>] can be derived as follows:
+
+- If $P_a \leq P_a$:
+$x = L \left(\frac{1}{\sqrt{P_a}} - \frac{1}{\sqrt{P_b}}\right)$
+$y = 0$
+
+- If $P_a \geq P_b$:
```

Comment:
Also shouldn't this be Po > Pb? 

in the code it is strictly greater and not greater than or equal
https://github.com/osmosis-labs/osmosis/blob/main/x/concentrated-liquidity/math/math.go#L196C8-L197